src/Spatial/Euclidean/Sphere3D.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Diagnostics.Contracts;
+using MathNet.Numerics.LinearAlgebra;
+using HashCode = MathNet.Spatial.Internals.HashCode;
+
+namespace MathNet.Spatial.Euclidean
+{
+    /// <summary>
+    /// Describes a 3 dimensional sphere.
+    /// </summary>
+    [Serializable]
+    public struct Sphere3D : IEquatable<Sphere3D>
+    {
+        /// <summary>
+        /// The center of the sphere.
+        /// </summary>
+        public readonly Point3D CenterPoint;
+        
+        /// <summary>
+        /// The radius of the sphere.
+        /// </summary>
+        public readonly double Radius;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Sphere3D"/>.
+        /// </summary>
+        /// <param name="centerPoint">The center of the Sphere</param>
+        /// <param name="radius">The radius of the Sphere</param>
+        public Sphere3D(Point3D centerPoint, double radius)
+        {
+            // Anypoint is same to other point.
+            if (radius <= 0)
+            {
+                throw new ArgumentException("The radius is negative.");
+            }
+
+            this.CenterPoint = centerPoint;
```

Comment:
Please remove all ```this.``` qualifiers

Reply:
IIRC we use StyleCop with default settings in this repro so if so `this.` is correct.

Reply:
I have already removed all "this." references in a previous [commit](https://github.com/mathnet/mathnet-spatial/commit/e3d3c8c469cb4fb639296b469663f266972e0328), and the CI build didn't complain, so I thought it was an acceptable change